File: python/docs_code_snippets/quickstart-tools.py
from opperai import Opper

opper = Opper()

# --- docs ---
result = opper.call(
    "weather-check",
    instructions="Use the available tools to help the user",
    input="What is the weather in Stockholm?",
    output_schema={
        "type": "object",
        "properties": {
            "answer": {"type": "string"},
        },
    },
    tools=[
        {
            "name": "get_weather",
            "description": "Get current weather for a city",
            "parameters": {
                "type": "object",
                "properties": {"city": {"type": "string"}},
                "required": ["city"],
            },
        }
    ],
)

print("Answer:", result.data)
# --- /docs ---

File: typescript/docs_code_snippets/quickstart-tools.ts
import { z } from "zod";
import { Opper } from "opperai";

const opper = new Opper();

const result = await opper.call("agent-round", {
  instructions: "Use the available tools to help the user",
  input: {
    message: "What is the weather in Stockholm?",
    tools: [
      {
        name: "get_weather",
        description: "Get current weather for a location",
        parameters: {
          type: "object",
          properties: { location: { type: "string" } },
          required: ["location"],
        },
      },
    ],
  },
  output_schema: z.object({
    message: z.string(),
    tool_call: z.object({
      name: z.string(),
      arguments: z.record(z.unknown()),
    }),
  }),
});

console.log(result.data);
